ci/lib.py

```diff
@@ -525,5 +542,28 @@ def test(data, path=None):
         host=controller, stream=True)
 
 
+def run_pep8_gate(nodes):
+    """
+    Run pep8 on controller node before running actual nodes
+    Fails CI if there are pep8 issues with the code
+
+    Args:
+        nodes (list): List of nodes received from duffy for CI
+    """
+    # get the controller node hostname, check note at the top of file
+    controller = nodes[4]
+
+    # setup node and install necessary packages
+    setup_controller(controller)
+
+    # sync controller with service code and patches applied on top
+    sync_controller(controller)
+
+    # CONTROLLER_WORK_DIR constant is the path where source code
+    # on controller is synced at, check constants.py and constant references
+    # run pep8 gate on pipeline service code
+    run_cmd("pep8 %s" % CONTROLLER_WORK_DIR, host=controller, stream=True)
```

Comment:
Can we use the `--exclude` switch to exclude certain files like `beanstalkc.py` and maybe others? I think we need to come up with a list of files that we don't want to lint here.

Reply:
@dharmit : done, added a config file where `beanstalkc.py` is excluded.